go/inst/rollback.go

```diff
@@ -0,0 +1,47 @@
+package inst
+
+// Main use case:
+// 1. User moves a group of slaves
+// 2. This hangs because of an unrelated issue
+// 3. User presses ^C to sent SIGINT
+// 4. Orchestrator tries to cleanup: e.g. start slaves.
+
+import (
+	"github.com/openark/golib/log"
+)
+
+type rollbackAction struct {
+	action   string
+	argument string
+}
+
+var rollbackActions []rollbackAction
+
+func AddRollbackAction(action string, argument string) {
+	act := rollbackAction{action, argument}
+	rollbackActions = append(rollbackActions, act)
+}
+
+func Rollback() {
+	if len(rollbackActions) == 0 {
+		return
+	}
+	log.Infof("Starting rollback")
+	for i, ra := range rollbackActions {
+		log.Infof("RollbackAction %d/%d (type: %s)", i+1, len(rollbackActions), ra.action)
+		switch ra.action {
+		case "print_message":
+			log.Infof("%s", ra.argument)
+		case "start_slave":
+			instanceKey, err := NewRawInstanceKey(ra.argument)
+			if err == nil {
+				log.Infof("Rollback: Starting replica %s", ra.argument)
+				StartSlave(instanceKey)
+			} else {
+				log.Debugf("Failed to get instanceKey for %s", ra.argument)
+			}
+		default:
+			log.Errorf("Unknown rollback action: %s", ra.action)
```

Comment:
I think doing a best effort here would be best. Try to restart replication where possible and log an error where this doesn't work. Then the admin or automation only has to fix a few machines instead of all machines.